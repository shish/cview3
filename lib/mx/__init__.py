""" mx Extension Series for Python

    Copyright (c) 1998-2000, Marc-Andre Lemburg; mailto:mal@lemburg.com
    Copyright (c) 2000-2001, eGenix.com Software GmbH; mailto:info@egenix.com
    See the documentation for further information on copyrights,
    or contact the author. All Rights Reserved.

    IMPORTANT:
    The subpackages included in the mx Extension series may have
    different license terms imposed on them. Be sure to read the
    documentation of each subpackage *before* using them.

"""
# mx Extension Series version number
# (note that subpackages have their own version number)
__version__ = '2.0'

# Copyright notice string
__copyright__ = """\
Copyright (c) 1998-2000, Marc-Andre Lemburg; mailto:mal@lemburg.com
Copyright (c) 2000-2001, eGenix.com Software GmbH; mailto:info@egenix.com
    See the documentation for further information on copyrights,
    or contact the author. All Rights Reserved.
"""

### Lazy import subpackages
#from mx.Misc import LazyModule

#ISO = LazyModule.LazyModule('ISO',locals(),globals())

#del LazyModule
