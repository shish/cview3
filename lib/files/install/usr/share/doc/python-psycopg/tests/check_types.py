# this script is a regression test for the psycopg type system. its output
# should be identical to the types.result files; any other result is an
# indication of an error in the psycopg type system.

import sys
from psycopg import *
from mx import DateTime

if len(sys.argv) > 1:
    DSN = sys.argv[1]
else:
    sys.stderr.write("Error: missing connection DSN\n")
    sys.exit(1)


class Infinity:
    def __init__(self, sign=1, typ='timestamp'):
        self.sign = sign
        self.type = typ
    def __str__(self):
        if self.sign < 0: return "%s '-infinity'" % self.type
        else: return "%s 'infinity'" % self.type
        
TYPES = (('name1', ('char(8)', 'aAbBcCdD')),
         ('name2', ('varchar(8)', 'aAbBcCdD')),
         ('name3', ('text', 'aAbBcCdD')),
         ('ivalue1', ('int8', 0)),
         ('ivalue2', ('int4', 256)),
         ('ivalue3', ('int2', -1)),
         ('fvalue1', ('float4', 0.0)),
         ('fvalue2', ('float8', 10e10)),
         ('boolean1', ('boolean', 't')),
         ('time1', ('time', Time(13,12,11))),
         ('date1', ('date', Date(1971,10,19))),
         ('datetime1', ('timestamp', Timestamp(2001, 10, 13, 7, 8, 9))),
         ('datetime2', ('timestamp', Infinity())),
         ('datetime3', ('timestamp', Infinity(-1))),
         ('interval1', ('interval', '2 days')),
         ('interval2', ('interval', '1 day 12:00:00')),
         ('interval3', ('interval', '6:00')),
         ('interval4', ('interval', '10 days 00:00:01')),
         ('interval5', ('interval', '1 year 2 months -1 day')),
         ('interval6', ('interval', '3 days 00:03:59.93')),
         ('binary1',  ('bytea', Binary(r'\\\001\002\003\004\\'))))
         
CHECKS = [NUMBER, LONGINTEGER, INTEGER, FLOAT, STRING, BOOLEAN,
          DATETIME, DATE, TIME, INTERVAL, BINARY, ROWID]


o = connect(DSN)
c = o.cursor()


## create the table
create = "CREATE TABLE types_test ("
for k, v in TYPES:
    create = create + "\n    %s %s, " % (k, v[0])
create = create[:-2] + "\n)"

c.execute(create)


## insert a row of data and a row of NULLs in the table
ins1 = ins2 = "INSERT INTO types_test VALUES ("
dict = {}
for k, v in TYPES:
    dict[k] = v[1]
    ins1 = ins1 + "%%(%s)s, " % k
    ins2 = ins2 + "NULL, "
ins1 = ins1[:-2] + ")"
ins2 = ins2[:-2] + ")"

c.execute(ins1, dict)
c.execute(ins2)


## now select from the tables and print data and types

def test(type):
    s = ''
    for t in CHECKS:
        if t == type: s = s + '1'
        else: s = s + '0'
    return s

c.execute("SELECT * FROM types_test")

row = c.fetchone() 
for i in range(len(row)):
    if c.description[i][1] == LONGINTEGER:
        print repr(row[i]), test(c.description[i][1])
    else:
        print str(row[i]), test(c.description[i][1])


## check for the row id (oid)

c.execute("SELECT oid FROM types_test")
row = c.fetchone()
print "oid", test(c.description[0][1])

#o.commit()
