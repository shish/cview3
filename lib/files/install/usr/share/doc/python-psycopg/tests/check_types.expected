aAbBcCdD 000010000000
aAbBcCdD 000010000000
aAbBcCdD 000010000000
0L 110000000000
256 101000000000
-1 101000000000
0.0 100100000000
100000000000.0 100100000000
1 000001000000
13:12:11.00 000000101000
1971-10-19 00:00:00.00 000000110000
2001-10-13 07:08:09.00 000000110000
999999-12-31 00:00:00.00 000000110000
-999998-01-01 00:00:00.00 000000110000
2:00:00:00.00 000000100100
1:12:00:00.00 000000100100
06:00:00.00 000000100100
10:00:00:01.00 000000100100
424:00:00:00.00 000000100100
3:00:03:59.92 000000100100
\\\001\002\003\004\\ 000000000010
oid 000000000001
